### ISSUE TITLE
 Unable to get azblob.StorageError for block blob operations

### ISSUE THREAD

**Issue by deiter** [Azure/azure-sdk-for-go#16781]
### Unable to get azblob.StorageError for block blob operations

**Steps to reproduce**
- Download the latest versions of github.com/Azure/azure-sdk-for-go
- Run simple block blob client and try to get properties for non-existing block blob:
```
import "github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
...
serviceClient, err := azblob.NewServiceClientWithSharedKey(primaryEndpoint, sharedKeyCredential, nil)
containerClient := serviceClient.NewContainerClient(containerName)
blobClient := containerClient.NewBlockBlobClient(blobName)
properties, err := blockBlob.GetProperties(ctx, nil)
log.Printf("error is %+v: %+v", reflect.TypeOf(err), err)
storageError := azblob.StorageError{}
log.Printf("error is azblob.StorageError: %+v", errors.As(err, &storageError))
```

**Expected results**
```
error is *azblob.StorageError: ....
error is azblob.StorageError: true
```

**Actual results: azblob.InternalError**
```
error is *azblob.InternalError: ===== RESPONSE ERROR (ErrorCode=BlobNotFound) =====
Description=, Details: (none)
error is azblob.StorageError: false
```

And I didn't find a way to get the original error code - azblob.StorageError / ErrorCode

Thank you!

**Comment by deiter** [comment]
Related issue: https://github.com/Azure/azure-storage-blob-go/issues/58

**Comment by zezha-msft** [comment]
@mohsha-msft I believe a relevant change just happened but it's not released yet, does it address this issue?

**Comment by tasherif-msft** [comment]
Hi @deiter!
Azure Core introduced https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/errors.go#L14 which can be cast using:
```
var responseErr *azcore.ResponseError
errors.As(err, &responseErr)
```
Let me know if solves the issue. 
I will close this in the meantime.

**Comment by jhendrixMSFT** [comment]
The usage of `*azcore.ResponseError` has been released in `azblob@v0.5.0`.
